backend/src/main/java/gov/cdc/usds/simplereport/service/PatientLinkService.java

```diff
@@ -106,4 +106,19 @@ public PatientLink createPatientLink(UUID testOrderUuid) {
     PatientLink pl = new PatientLink(to);
     return plrepo.save(pl);
   }
+
+  public String getObfuscatedPatientNameFromLink(UUID internalId)
+      throws ExpiredPatientLinkException {
+    PatientLink link = getPatientLink(internalId);
+    TestOrder to = link.getTestOrder();
+    Person p = to.getPatient();
+
+    contextHolder.setContext(link, to, p);
+
+    if (link.isExpired()) {
```

Comment:
i would move this to line 113, there is no need to load up the test order or the patient if the link is invalid

Reply:
See [ee17213](https://github.com/CDCgov/prime-simplereport/pull/3305/commits/ee1721366213e572f729a5704dae4622c109075d) 👏